cmd/tempo-vulture/main.go

```diff
@@ -148,12 +143,53 @@ func main() {
 	log.Fatal(http.ListenAndServe(prometheusListenAddress, nil))
 }
 
+func intervalsBetween(start, stop time.Time, interval time.Duration) []time.Time {
+	if stop.Before(start) {
+		return nil
+	}
+
+	intervals := []time.Time{start}
+	next := start.Add(interval)
+
+	for next.Before(stop) {
+		intervals = append(intervals, next)
+		next = next.Add(interval)
+	}
+
+	return intervals
+}
+
+func trimOutdatedIntervals(intervals []time.Time, lookBehind time.Duration) []time.Time {
```

Comment:
would be clearer to just choose the starttime and use the above method. i don't think we need this method.